File: chain_LLM/chain.js
import {config} from "dotenv";
config();

import {ChatGoogleGenerativeAI} from "@langchain/google-genai";
import { LLMChain } from "langchain/chains";
import {PromptTemplate} from "@langchain/core/prompts";

// 1. setup the model
const model = new ChatGoogleGenerativeAI({
    model: "models/gemini-2.5-flash",
    maxOutputTokens: 2048,
    temperature: 0.7,
    apiKey: process.env.GOOGLE_API_KEY,
});

// 2. create a prompt
const prompt = PromptTemplate.fromTemplate(
    "Explain the concept of {topic} to a begineer."
);

// 3. Build a chain
const chain = new LLMChain({
    llm: model,
    prompt: prompt,
})

// 4. Run it
const res = await chain.run("qunatum computing");
console.log("Gemini Response:\n",res);

File: chain_LLM/chain.py
from dotenv import load_dotenv
import os

load_dotenv()
api_key = os.getenv("GOOGLE_API_KEY")

# import and setup the MODEL
from langchain_google_genai import ChatGoogleGenerativeAI
from langchain.prompts import PromptTemplate
from langchain.chains import LLMChain

# create Gemini model
model = ChatGoogleGenerativeAI(
    model="models/gemini-2.0-flash",
    temperature =0.7,
    max_output_tokens=2048,
    google_api_key=api_key
    
)

#Create the prompt
prompt = PromptTemplate.from_template("Explain the concept of {topic} to a begineer.")

#Build chain
chain=LLMChain(
    llm=model,
    prompt=prompt
)

response = chain.run("quantum computing")
print("Gemini Response:\n",response)